pkg/pss/utils.go

```diff
@@ -1,125 +1,52 @@
 package pss
 
 import (
-	"strings"
+	"fmt"
 
 	kyvernov1 "github.com/kyverno/kyverno/api/kyverno/v1"
 	"github.com/kyverno/kyverno/pkg/utils"
 	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/pod-security-admission/policy"
 )
 
-func containsContainer(containers interface{}, containerName string) bool {
-	switch v := containers.(type) {
-	case []interface{}:
-		for _, container := range v {
-			switch v := container.(type) {
-			case corev1.Container:
-				if v.Name == containerName {
-					return true
-				}
-			case corev1.EphemeralContainer:
-				if v.Name == containerName {
-					return true
-				}
-			}
-		}
-	case []corev1.Container:
-		for _, container := range v {
-			if container.Name == containerName {
-				return true
-			}
-		}
-	case []corev1.EphemeralContainer:
-		for _, container := range v {
-			if container.Name == containerName {
-				return true
-			}
-		}
+// getPodWithMatchingContainers extracts matching container/pod info by the given exclude rule
+// and returns pod manifests containing spec and container info respectively
+func getPodWithMatchingContainers(exclude kyvernov1.PodSecurityStandard, pod *corev1.Pod) (podSpec, matching *corev1.Pod) {
+	if len(exclude.Images) == 0 {
+		podSpec = &corev1.Pod{}
```

Comment:
What is this code doing ?

Reply:
It copies the pod's metadata and spec to a new object except `containers`, `initContianers` and `ephemeralContainers`, the new object is used to validate PSS controls at pod level.

Reply:
That's a bit hard to read IMHO.
Why not deep copy, then set to `nil` instead of empty array ?
It would make things easier to follow I guess.

Reply:
Sure we can do that. Sending the change.

Reply:
Done.